src/main/kotlin/ecommerce/controller/product/ProductController.kt

```diff
@@ -20,18 +21,18 @@ import java.net.URI
 class ProductController(private val productService: ProductService) {
```

Comment:
```suggestion
@RequestMapping("/products")
class ProductController(private val productService: ProductService) {
    // implementation

    @GetMapping("/{id}")
    fun getProductById(
        @PathVariable("id") id: Long,
    ): ResponseEntity<ProductResponse> {
        // implementation
    }
    
    @PutMapping("/{id}")
    fun updateProduct(
        @PathVariable("id") id: Long,
        @Valid @RequestBody request: UpdateProductRequest,
    ): ResponseEntity<ProductResponse> {
        // implementation
    }
}
```

You can use `@RequestMapping` like this, since all the APIs share the same `/products` prefix.

Reply:
Thank you! that makes the endpoints clear and short!